src/linearmixedmodel.jl

```diff
@@ -920,6 +943,9 @@ function zerocorr!(m::LinearMixedModel{T}, trmns) where {T}
             zerocorr!(trm)
         end
     end
+    newparmap = mkparmap(reterms)
+    copyto!(m.parmap, newparmap)
+    resize!(m.parmap, length(newparmap))
```

Comment:
This dance is necessary because `LinearMixedModel` is an immutable struct, right? (Not suggesting a change, just double checking my understanding.)

Reply:
Exactly